src/EventStore.Core/TransactionLog/FileNamingStrategy/IFileNamingStrategy.cs

```diff
@@ -1,9 +1,16 @@
+using System;
+
 namespace EventStore.Core.TransactionLog.FileNamingStrategy {
 	public interface IFileNamingStrategy {
 		string GetFilenameFor(int index, int version);
 		string DetermineBestVersionFilenameFor(int index);
 		string[] GetAllVersionsFor(int index);
 		string[] GetAllPresentFiles();
+		void EnumerateAllFiles(
+			Func<string, int> getNextFileNumber,
+			Action<string, int, int> onLatestVersionFound = null,
+			Action<string, int> onOldVersionFound = null,
+			Action<string, int> onFileMissing = null);
```

Comment:
Prefer not to use optional parameters

Reply:
Fixed in 3c1b84b61a4c4db2d03f367d0f74a1fa12106881